app/controllers/settings_controller.rb

```diff
@@ -0,0 +1,18 @@
+class SettingsController < ApplicationController
+  def edit
+  end
+
+  def update
+    if current_user.update(user_params)
+      redirect_to root_path, notice: "Profile updated successfully."
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation)
```

Comment:
I think @Shpigford wants also to update the family name, right?

Reply:
This is correct. Need to add the family name to this form as well.